InterGalacticCurrencyConvertor/src/main/java/statement/MaterialCostStatement.java

```diff
@@ -0,0 +1,59 @@
+package main.java.statement;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+import main.java.DataProcessResult;
+import main.java.DataProcessor;
+import main.java.numerals.InterGalacticNumerals;
+import main.java.materials.Materials;
+import main.java.numerals.RomanNumerals;
+
+/**
+ * Processes statement of the form "x m is d credits" where "x" denotes a number of
+ * InterGalactic numerals separated by white-spaces, "m" is the material name and
+ * "d" is the decimal value equivalent to "x" amount of material "m".
+ */
+public class MaterialCostStatement implements Statement {
```

Comment:
Consider using a regex for parsing - it may make the code simpler.

Reply:
updated